Lib/_collections_abc.py

```diff
@@ -409,6 +415,81 @@ def __subclasshook__(cls, C):
         return NotImplemented
 
 
+class _CallableGenericAlias(GenericAlias):
+    """ Internal class specifically for consistency between the ``__args__`` of
+    ``collections.abc.Callable``'s and ``typing.Callable``'s ``GenericAlias``.
+
+    See :issue:`42195`.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if not isinstance(args, tuple) or len(args) != 2:
+            raise TypeError("Callable must be used as "
+                            "Callable[[arg, ...], result].")
+        _typ, _args = args
+        if not isinstance(_args, (tuple, EllipsisType)) or len(_args) != 2:
+            raise TypeError("Callable must be used as "
+                            "Callable[[arg, ...], result].")
```

Comment:
Let's make sure the two error messages are not identical. Here I would call out that it can either be a list of types or an ellipsis.